overlays/emacs.nix

```diff
@@ -86,13 +105,22 @@ let
     withWebP = true;
     nativeComp = true;
   };
-
+  
   emacsPgtk = mkPgtkEmacs "emacs-pgtk" ../repos/emacs/emacs-master.json { withSQLite3 = true; withGTK3 = true; };
 
   emacsPgtkNativeComp = mkPgtkEmacs "emacs-pgtk-native-comp" ../repos/emacs/emacs-master.json { nativeComp = true; withSQLite3 = true; withGTK3 = true; };
 
   emacsUnstable = (mkGitEmacs "emacs-unstable" ../repos/emacs/emacs-unstable.json { });
 
+  emacsGitTreeSitter = super.lib.makeOverridable (mkGitEmacs "emacs-git-tree-sitter" ../repos/emacs/emacs-feature_tree-sitter.json) {
+    withTreeSitter = true;
+    withTreeSitterPlugins = (plugins: with plugins; [
```

Comment:
Why didn't you also add `tree-sitter-nix` by default?
BTW great job.

Reply:
Because currently only these three major modes have tree-sitter support.

Reply:
`ts-mode.el` was merged into the tree-sitter branch the other day (https://git.savannah.gnu.org/cgit/emacs.git/commit/?id=45b8204e09b7d399b792bb26c799efe48835c4a7), so maybe we can add the TypeScript module as well.